### PR TITLE
Overhaul of `ControlMonitoringStatus` API

### PR DESCRIPTION
**Pull request by oxisto**
This PR is a complete overhaul of the `ControlMonitoringStatus` API, which was flawed, broken and incomplete and hard to understand. The following things changed:
- `ControlMonitoringStatus` was renamed to `ControlInScope`, which is a better description of what it is. It is a control that is in scope of a ToE
- Added RPC calls to add and remove controls to/from the scope of a ToE
- Adjusted event hooks that are connected to the API to also differentiate between update/remove/add

### CODE REVIEW

File: service/orchestrator/toe.go [hunk 1]
```diff
@@ -221,26 +221,90 @@ func (svc *Service) UpdateControlMonitoringStatus(ctx context.Context, req *orch
 		return nil, service.ErrPermissionDenied
```
Comment (anatheka) [inline service/orchestrator/toe.go:221]: Shouldn't you return a status error?

File: service/orchestrator/toe.go [hunk 2]
```diff
@@ -200,16 +200,16 @@ func (svc *Service) ListControlMonitoringStatus(ctx context.Context, req *orches
...
```
Comment (anatheka) [inline service/orchestrator/toe.go:200]: Shouldn't you return a status error?

File: service/orchestrator/toe.go [hunk 3]
```diff
@@ -221,26 +221,90 @@ func (svc *Service) UpdateControlMonitoringStatus(ctx context.Context, req *orch
...
+func (svc *Service) UpdateControlInScope(ctx context.Context, req *orchestrator.UpdateControlInScopeRequest) (res *orchestrator.ControlInScope, err error) {
+	// Validate request
+	err = service.ValidateRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check, if this request has access to the cloud service according to our authorization strategy.
+	if !svc.authz.CheckAccess(ctx, service.AccessRead, req) {
+		return nil, service.ErrPermissionDenied
```
Comment (anatheka) [inline service/orchestrator/toe.go:249]: Shouldn't you return a status error?

File: service/orchestrator/toe.go [hunk 4]
```diff
@@ -221,26 +221,90 @@ func (svc *Service) UpdateControlMonitoringStatus(ctx context.Context, req *orch
...
+func (svc *Service) RemoveControlFromScope(ctx context.Context, req *orchestrator.RemoveControlFromScopeRequest) (res *emptypb.Empty, err error) {
+	// Validate request
+	err = service.ValidateRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check, if this request has access to the cloud service according to our authorization strategy.
+	if !svc.authz.CheckAccess(ctx, service.AccessRead, req) {
+		return nil, service.ErrPermissionDenied
```
Comment (anatheka) [inline service/orchestrator/toe.go:285]: see above

**Review by anatheka — COMMENTED** [review]
Only a minor issue.

File: service/orchestrator/toe.go (hunk 2 above)
Comment (oxisto) [reply service/orchestrator/toe.go:200]: Ah lol! `var ErrPermissionDenied = status.Errorf(codes.PermissionDenied, "access denied")`

**Review by anatheka — APPROVED** [review]
LGTM

**Pull request merged**